activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -375,7 +375,7 @@ def construct_relation_for_association_calculations
         # FIXME: as far as I can tell, `from` will always be an Arel::Table.
         # There are no tests that test this branch, but presumably it's
         # possible for `from` to be a list?
-        apply_join_dependency(self, construct_join_dependency(from))
+        apply_join_dependency(self, construct_join_dependency(Array.wrap(from)))
```

Comment:
I think this makes sense. @tenderlove you added the comment above, any feedback here?